app/stats/Metrics.tsx

```diff
@@ -96,13 +98,38 @@ const MetricBox = ({
     data: samples.map((s) => s.label),
     scaleType: "band",
   };
+
+  const calculateTrend = (data: Array<number>) => {
+    const lastValue: number = data[data.length - 1];
+    const penultimateValue: number = data[data.length - 2];
+    const trend = ((lastValue - penultimateValue) / penultimateValue) * 100;
+    const twoDecimalsTrend = Math.round(trend * 100) / 100;
+    return twoDecimalsTrend;
+  };
+
+  const trend = calculateTrend(data);
+
+  const shouldBeGreen = (metric: MetricDefinition, trend: number) => {
+    return (
+      (trend >= 0 && metric.positiveDirection === "moreIsBetter") ||
+      (trend <= 0 && metric.positiveDirection === "lessIsBetter")
+    );
+  };
+
+  const isGreen: boolean = shouldBeGreen(metric, trend);
```

Comment:
Tady je ta typová anotace zbytečná. Ono ale není moc jasné, kdy jsou ty volitelné anotace potřeba a kdy ne, že? Obecně asi dává smysl tu volitelnou anotaci přidat, když to zlepší čitelnost nebo když je slušná šance, že překladač odvodí ten typ jinak a ten rozdíl tě pak upozorní na chybu. Tady bych klidně vynechal.